src/main/java/com/example/ninthhomework/controller/CharacterListController.java

```diff
@@ -23,20 +23,25 @@ public CharacterListController(CharactersService charactersService) {
 
     //IDも含めて全て返す
     @GetMapping("/characters/all")
-    public List<Characters> charactersServices() {
+    public List<Characters> selectAllCharacters() {
         return charactersService.getCharacters().stream().toList();
     }
 
+    //指定したIDの内容のみ返す
+    @GetMapping("/characters/{id}")
+    public Characters findCharacterById(@PathVariable("id") int id) {
+        return charactersService.findById(id);
+    }
 
     //IDは含めずに名前と年齢のみ返す
     @GetMapping("/characters")
-    public List<CharactersResponse> charactersResponse() {
+    public List<CharactersResponse> selectResponseCharacters() {
```

Comment:
レスポンスするのは自明なので
```suggestion
    public List<CharactersResponse> selectCharacters() {
```
でよいかと思います。

Reply:
修正しました

Reply:
fe2eae21082c1fd26fa51ef34f0a6c8cd6da5ce7
修正しました